odxtools/basicstructure.py

```diff
@@ -66,9 +68,8 @@ def coded_const_prefix(self, request_prefix: bytes = b'') -> bytes:
         prefix = b''
         encode_state = EncodeState(prefix, parameter_values={}, triggering_request=request_prefix)
         for p in self.parameters:
-            if isinstance(p, CodedConstParameter) and p.bit_length % 8 == 0:
-                encode_state.coded_message = p.encode_into_pdu(encode_state)
-            elif isinstance(p, MatchingRequestParameter):
+            if isinstance(p, (CodedConstParameter, NrcConstParameter, MatchingRequestParameter,
+                              PhysicalConstantParameter)):
```

Comment:
Use `p.is_settable` instead?

Reply:
that would be not correct from a logical POV: what is required here is to determine the "constant prefix" of a structure (i.e., the part which allows to identify it), i.e., all parameters which are constant -- meaning that they are defined statically or only depending on previous communication (`MatchingRequestParameter`). The problem is that some parameters parameters are not settable, but can still not be used for identification (e.g. `ReservedParameter`)

maybe we can introduce a new property ('.is_prefixable`?) for this purpose, but I don't think that the complexity costs of this would be currently worth it...

Reply:
The way I see it, a `ReservedParameter` should also be part of the prefix, any other examples?

Reply:
`ReservedParameter` cannot be made part of the prefix because its purpose is to specify that the data it covers that is send at runtime is undefined. (possibly, we can make it settable, but that would be rather inelegant..)

another example is (maybe) `SystemParameter`, but this one is currently a stub and throws `NotImplementedError` upon accessing `.is_settable` (as do a few others...)

Reply:
I have seen cases where `Suppress Positive Response Message Indication Bit` was implemented as a `ReservedParameter`.

I don't see a reason to explicitly exclude it.

Reply:
If that's the case, then the logic would be fine for this PR. 

As a side note, the logic you have now in ReservedParameter raise an error for non zero values when decoding, that doesn't match the statement of "The file format does not specify any value for such parameters"

Reply:
hm, that got me scratching my head, and I thus looked up the spec. (cf p. 78 and 79): It says that all bits shall be ignored, but it also says that all its bits shall be set to zero, which IMO is a contradiction. I suppose that for the purpose of encoding reserved parameters, all its bits should be set to 0 (because what else can we do?) whilst during decoding these parameters ought to be ignored. I think that the words of the spec can be interpreted that way, but it is formulated quite confusingly. Agree?

(In the mean time, I have added a patch to the series which does just that: bd853f5c91504cf1bf7105020fc29fbeccd4b626)